tests/modules/apigee_mock_client/src/ApigeeMockClientServiceProvider.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_mock_client;
+
+use Drupal\Core\DependencyInjection\ServiceProviderBase;
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Class ApigeeM10nTestServiceProvide
+ *
+ * This class is automatically picked up by the container builder.
+ * See: https://www.drupal.org/docs/8/api/services-and-dependency-injection/altering-existing-services-providing-dynamic-services
+ *
+ * @package Drupal\apigee_mock_client
+ */
+class ApigeeMockClientServiceProvider extends ServiceProviderBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function alter(ContainerBuilder $container) {
+    // Override the ClientFactory with our mock client factory.
+    $container->getDefinition('apigee_edge.sdk_connector')
```

Comment:
Why don't you just created a service decorator and DI-d your client factory?
https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/tests/modules/apigee_edge_test/apigee_edge_test.services.yml#L20
https://github.com/apigee/apigee-edge-drupal/blob/8.x-1.x/tests/modules/apigee_edge_test/src/SDKConnector.php

Reply:
That would be another way to do this, but is it necessarily better?

Reply:
Yes, because what you have DI-d you can easily replace when you write tests. Where you used `\Drupal::service()` instead you can not mock that easily.

Reply:
Gotcha, I'll add a todo, but I don't think we need to get this in for this pull request.